### PR TITLE
Adds ability to have catch-all triggers on groups

### PR DESCRIPTION
**Pull request by nicpottier**
IE, you can say for messages that aren't handled for a particular group do one thing, for other groups, do another.

### CODE REVIEW

File: temba/triggers/views.py [hunk 1]
```diff
@@ -638,25 +663,32 @@ def form_valid(self, form):
...
 
         def form_valid(self, form):
             user = self.request.user
             org = user.get_org()
+            groups = form.cleaned_data['groups']
 
-            # first archive all catch all message triggers
-            Trigger.objects.filter(org=org,
+            # first archive all catch all message triggers with matching groups
+            Trigger.objects.filter(org=org, groups=groups,
```
Comment (norkans7) [inline temba/triggers/views.py]: I think here we want to use groups__in to match all triggers with any of the groups

Comment (nicpottier) [inline temba/triggers/views.py]: Good catch, fixed and tested.

**Comment by ericnewcomer** [comment]
What happens if the group is deleted?

**Comment by nicpottier** [comment]
Hrmm, good question..

We don't delete groups, just make them inactive, so it will still be linked but the UI when editing will be goofy. The next time you edit it, the group won't appear and when you try to save it'll fail if there is another catch-all that operates on every contact.

**Comment by ericnewcomer** [comment]
Should we maybe archive any triggers based on groups when the user deletes them?

**Comment by nicpottier** [comment]
Ya if it is the only group for that trigger that probably makes sense.

**Comment by nicpottier** [comment]
Now marks triggers that operate only on a released group as inactive. (not just archived because we don't want to allow them to unarchive them)

**Comment by norkans7** [comment]
Yep

**Pull request merged**